packages/bsky/src/auto-moderator/abyss.ts

```diff
@@ -0,0 +1,103 @@
+import axios from 'axios'
+import { CID } from 'multiformats/cid'
+import * as ui8 from 'uint8arrays'
+import { resolveBlob } from '../api/blob-resolver'
+import { retryHttp } from '../util/retry'
+import { PrimaryDatabase } from '../db'
+import { IdResolver } from '@atproto/identity'
+import { labelerLogger as log } from '../logger'
+
+export interface TakedownFlagger {
+  scanImage(did: string, cid: CID): Promise<string[]>
+}
+
+export class Abyss implements TakedownFlagger {
+  protected auth: string
+
+  constructor(
+    public endpoint: string,
+    protected password: string,
+    public ctx: { db: PrimaryDatabase; idResolver: IdResolver },
+  ) {
+    this.auth = basicAuth(this.password)
+  }
+
+  async scanImage(did: string, cid: CID): Promise<string[]> {
+    const res = await retryHttp(async () => {
+      try {
+        return await this.makeReq(did, cid)
+      } catch (err) {
+        log.warn(
+          { err, did, cid: cid.toString() },
+          'blob scanner request failed',
+        )
+        throw err
+      }
+    })
+    return this.parseRes(res)
+  }
+
+  async makeReq(did: string, cid: CID): Promise<ScannerResp> {
+    const { stream, contentType } = await resolveBlob(
+      did,
+      cid,
+      this.ctx.db,
+      this.ctx.idResolver,
+    )
+    const { data } = await axios.post(this.getReqUrl({ did }), stream, {
```

Comment:
Things we might want to do at this HTTP request call site:

- log response status code and duration (ms). we're also doing logging+metrics in abyss, but good to do it on this side of the loadbalancer as well
- have a timeout. 10sec?

Reply:
yup both seem reasonable 👍